pretyping/retyping.mli

```diff
@@ -31,8 +31,11 @@ val get_type_of :
 val get_sort_of :
   ?polyprop:bool -> env -> evar_map -> types -> Sorts.t
 
+(* When [truncation_style] is [true], tells if the type has been explicitly
+   truncated to Prop or (impredicative) Set; in particular, singleton type and
+   small inductive types, which have all eliminations to Type, are in Type *)
 val get_sort_family_of :
-  ?polyprop:bool -> env -> evar_map -> types -> Sorts.family
+  ?truncation_style:bool -> ?polyprop:bool -> env -> evar_map -> types -> Sorts.family
```

Comment:
I don't like using a `bool` for this.
How about

- (bikeshed the names as you wish)
```ocaml
type template_is_in_type = TemplateAlwaysInType | TemplateCanBeSmall
```

- also considered this one, it's kinda nicely generic yet at the same time passing anything except `None` or `Some InType` makes no sense
```ocaml
?template_family:Sorts.family (* if given, template polymorphic types are considered to be in this family *)
```